lib/rack/test.rb

```diff
@@ -30,7 +30,14 @@ module Rack
   module Test
     # The default host to use for requests, when a full URI is not
     # provided.
-    DEFAULT_HOST = 'example.org'.freeze
+    def self.default_host
+      @@default_host ||= 'example.org'.freeze
```

Comment:
Should we consider the following implementation:

```
DEFAULT_HOST = 'example.org'.freeze
class << self
  @default_host = DEFAULT_HOST
  attr_accessor :default_host
end
```

I'm not sure if we can remove `DEFAULT_HOST`.

Consider that `||=` isn't thread safe and we could in theory have multiple threads accessing this field.

We've followed this model elsewhere: https://github.com/rack/rack/blob/main/lib/rack/request.rb#L17-L41

Don't have a strong opinion about it, just wanted to make a suggestion.

Reply:
We definitely cannot accept the use of class variables.  Class variables in Ruby are broken and should never be used.

This approach:

```ruby
class << self
  @default_host = DEFAULT_HOST
  attr_accessor :default_host
end
```

Is broken.  It sets the instance variable in the singleton class instead of the class.  However, with that fixed, because `Rack::Test` is a module and not a class, the approach seems acceptable (it wouldn't be acceptable for a class, but then it breaks in a subclass unless you copy the instance variable during subclassing).

Reply:
Yeah using the approach above failed 1-44 tests in a somewhat random nature.  Moving to this:
```
@default_host = DEFAULT_HOST
class << self
  attr_accessor :default_host
end
```
makes all the tests passing again.  Since it is a module, I am not sure I see the difference.